subsys/bluetooth/host/conn_internal.h

```diff
@@ -269,12 +262,85 @@ struct bt_conn {
 		uint16_t subversion;
 	} rv;
 #endif
+
+	/* Callback into the higher-layers (L2CAP / ISO) to return a buffer for
+	 * sending `amount` of bytes to HCI.
+	 *
+	 * Scheduling from which channel to pull (e.g. for L2CAP) is done at the
+	 * upper layer's discretion.
+	 */
+	struct net_buf * (*tx_data_pull)(struct bt_conn *conn, size_t amount);
+
+	/* Get (and clears for ACL conns) callback and user-data for `buf`. */
+	void (*get_and_clear_cb)(struct bt_conn *conn, struct net_buf *buf,
+				 bt_conn_tx_cb_t *cb, void **ud);
+
+	/* Return true if upper layer has data to send over HCI */
+	bool (*has_data)(struct bt_conn *conn);
+
+	/* For ACL: List of data-ready L2 channels. Used by TX processor for
+	 * pulling HCI fragments. Channels are only ever removed from this list
+	 * when a whole PDU (ie all its frags) have been sent.
+	 */
+	sys_slist_t		l2cap_data_ready;
```

Comment:
In this case of ISO broadcaster builds this field is unused, right? Should we guard it?

Reply:
can we keep that for later? Don't want to anger the build system gods again.

Reply:
I'm OK with doing optimizations later :)